assembly/runtime/promise.ts

```diff
@@ -0,0 +1,138 @@
+import { util } from "./util";
+import { env } from "./env";
+import { u128 } from ".";
+
+function encodeArgs<T>(t: T) : Uint8Array {
+  if (t instanceof Uint8Array) {
+    return t;
+  }
+  return encode<T>(t);
+}
+
+// TODO: not sure if these types make any sense at all
```

Comment:
seems like this was ok after all?

Reply:
Yeah but perhaps we should make a types file so there is some reuse.  I think we can pass on that in this PR.

Reply:
added TODO comment ✅